Source/WebCore/html/parser/HTMLParserScheduler.cpp

```diff
@@ -120,6 +120,9 @@ bool HTMLParserScheduler::shouldYieldBeforeExecutingScript(const ScriptElement*
     if (UNLIKELY(m_documentHasActiveParserYieldTokens))
         return true;
 
+    if (document->view() && !document->view()->hasEverPainted() && document->isLayoutPending())
+        return true;
+
```

Comment:
I wonder if we should also A/B test moving the "if (scriptElement)" block above this line. Forcing layout just because we parse an async or defer script tag seems wrong. Forcing layout before we execute a short inline script seems like an experimental question.

Reply:
That is a good point, I will schedule that A/B test.

Thanks for reviewing!